crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -127,7 +127,12 @@ impl IRGenerator {
 
     pub fn codegen_main(&mut self) -> Result<(), RuntimeError> {
         let main_body = self.program.take_main_body();
-        self.codegen_expression(&main_body)?;
+        let value = self.codegen_expression(&main_body)?;
+        let node_ids = value.to_node_ids();
+
+        if self.program.main().return_type != Type::Unit {
```

Comment:
I don't think it is needed, if you return an empty list it should still work.

Reply:
If it should work, there is perhaps a bug in SSA

Reply:
I don't know, but if it does not work with an empty list, then I am fine with checking for it here.